hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/buffer/BackupMessagesLoader.java

```diff
@@ -96,19 +103,26 @@ public void loadMessages(MessageRepository messageRepository) {
 
     private void sendMessages(List<BackupMessage> messages) {
         logger.info("Sending {} messages from backup storage.", messages.size());
+        ExecutorService executor = Executors.newFixedThreadPool(16);
 
         int sentCounter = 0;
         int discardedCounter = 0;
         for (BackupMessage backupMessage : messages) {
             Message message = new JsonMessage(backupMessage.getMessageId(), backupMessage.getData(), backupMessage.getTimestamp());
-            Optional<Topic> topic = loadTopic(fromQualifiedName(backupMessage.getQualifiedTopicName()));
+            Optional<Topic> topic = loadTopic(fromQualifiedName(backupMessage.getQualifiedTopicName()), executor);
             if (sendMessageIfNeeded(message, topic, "sending")) {
                 sentCounter++;
             } else {
                 discardedCounter++;
             }
         }
 
+        executor.shutdownNow();
```

Comment:
shouldn't we free those resources in some finally block just to be on the safe side?